apps/native/app/src/ui/lib/card/air-discount-card.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react'
+import { FormattedMessage, useIntl } from 'react-intl'
+import Clipboard from '@react-native-clipboard/clipboard'
+import styled, { useTheme } from 'styled-components/native'
+
+import { dynamicColor } from '../../utils'
+import { font } from '../../utils/font'
+import { Typography } from '../typography/typography'
+
+const Host = styled.View`
+  margin-vertical: ${({ theme }) => theme.spacing[1]}px;
+  border-radius: ${({ theme }) => theme.border.radius.large};
+  border-width: ${({ theme }) => theme.border.width.standard}px;
+  border-color: ${dynamicColor(
+    ({ theme }) => ({
+      light: theme.color.blue200,
+      dark: theme.shades.dark.shade300,
+    }),
+    true,
+  )};
+`
+const Content = styled.View`
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  padding: ${({ theme }) => theme.spacing[2]}px;
+  padding-right: ${({ theme }) => theme.spacing[1]}px;
+  padding-bottom: ${({ theme }) => theme.spacing[1]}px;
+`
+
+const Message = styled.View`
+  align-items: flex-start;
+  flex: 1;
+`
+
+const Right = styled.View`
+  align-items: flex-end;
+`
+
+const CopyCodeContainer = styled.View`
+  border-top-width: ${({ theme }) => theme.border.width.standard}px;
+  border-top-color: ${dynamicColor(
+    ({ theme }) => ({
+      light: theme.color.blue200,
+      dark: theme.shades.dark.shade300,
+    }),
+    true,
+  )};
+`
+
+const CopyCodeButton = styled.TouchableOpacity`
+  align-items: center;
+  justify-content: center;
+  padding: ${({ theme }) => theme.spacing[2]}px;
+`
+
+const ValidUntil = styled.Text`
```

Comment:
Can you use Typography component here?

Reply:
No, or at least did not find a way to make it work. The thing is that this `ValidUntil` text has different colors for light/dark theme than the regular texts. The Typography component can receive a color prop but does not support values on the format of dark/light mode.  So we would need to do a change in the Typography component for that to work and I decided that was out of scope for this, since it was only this one place.